python-frontend/src/main/java/org/sonar/plugins/python/api/symbols/Symbol.java

```diff
@@ -35,6 +36,13 @@
 
   Kind kind();
 
+  /**
+   * Returns fully qualified name of the type if any
+   */
+  @Beta
+  @CheckForNull
+  String annotatedTypeName();
```

Comment:
As discussed, I think it would be nice to add a unit test (for both `annotatedTypeName` and `annotatedReturnTypeName`) for ProjectLevelSymbolTable to assert we keep information about annotated types.
Something similair to what we did for metaclasses

https://github.com/SonarSource/sonar-python/blob/d299c45234efe70567244369e9d813d7dcdc839e/python-frontend/src/test/java/org/sonar/python/semantic/ProjectLevelSymbolTableTest.java#L266